lib/options.nix

```
@@ -366,4 +366,108 @@ rec {
         else
           example;
     };
+
+  mkLazyLoadOption =
+    {
+      originalName,
+      lazyLoadDefaults ? { },
+    }:
+    let
+      pluginDefault = {
+        enable = false;
+      } // lazyLoadDefaults;
+    in
+    mkOption {
+      description = ''
+        Lazy-load settings for ${originalName}.
+      '';
+      type =
+        with helpers.nixvimTypes;
+        let
+          triggerType = oneOf [
+            rawLua
+            str
+            (listOf str)
+          ];
+        in
+        submodule {
+          options = with defaultNullOpts; {
+
+            enable = mkOption {
+              type = bool;
+              default = pluginDefault.enable;
+              description = ''
+                Enable lazy-loading for ${originalName}
+              '';
+            };
+
+            # Spec loading:
+            enabled = mkStrLuaFnOr bool pluginDefault.enabledInSpec or null ''
+              When false, or if the function returns false, then ${originalName} will not be included in the spec.
+              Equivalence: lz.n => enabled; lazy.nvim => enabled
```

```suggestion
              When false, or if the function returns false, then ${originalName} will not be included in the spec.

              Equivalence: lz.n => enabled; lazy.nvim => enabled
```

I think this would benefit from being in its own paragraph.

Note, in commonmark, a single linebreak is _not_ rendered in the output. You either need a blankline to cause a paragraph-break, or a `\` at the end of a line to treat the following newline as a line-break.